### ISSUE TITLE
Regression in the text formatting of 'handle_analyzer_status'

### ISSUE THREAD

**Issue by museun** [rust-analyzer/rust-analyzer#4115]
While looking to see which 'commands'/'actions' were causing RA to stall, the 'Rust Analyzer: Status' output has regressed.
The requests output used to print out each request per line, now it just puts them on a single line.

The change happened here. 
https://github.com/rust-analyzer/rust-analyzer/commit/311cbbdad599d51c6f08f7dd72c299f7c0128bb2#diff-cb373ef102a169ef5e1af6f370fd21ccR53
The change from `writeln!()` to`stdx::format_to!()`. `format_to!()` uses `write!()` internally. I don't know if this has caused other parts of the addon's output to change, but using it in a loop here definitely has.

Current output, showing 'requests' not being separated onto their own line.
```text
workspaces:
138 packages loaded

analysis:
2877 (31mb) files
142854 (13mb) symbols
1626 trees, 128 retained
26092 trees, 128 retained (macros)


memory:
0b allocated 0b resident
gc 948 seconds ago

requests:    2 textDocument/codeAction             42087ms    3 textDocument/codeAction             1ms    4 textDocument/codeAction             1ms  673 textDocument/hover                  5ms  678 textDocument/codeAction             0ms
```

**Comment by lnicola** [comment]
There's one here, too: https://github.com/rust-analyzer/rust-analyzer/commit/311cbbdad599d51c6f08f7dd72c299f7c0128bb2#diff-b99274f974e29549262969da548dde9cR43.

**Comment by flodiebold** [comment]
Should be fixed by #4256.
